src/Test/Utilities/Portable/Compilation/ControlFlowGraphVerifier.cs

```diff
@@ -45,76 +48,365 @@ public static string GetFlowGraph(Compilation compilation, ControlFlowGraph grap
-                    stringBuilder.Builder.AppendLine($"    Jump if {(block.Conditional.JumpIfTrue ? "True" : "False")} to Block[{index}]");
+                    int index = conditionalBranch.Destination.Ordinal;
+                    Assert.Same(blocks[index], conditionalBranch.Destination);
+                    appendLine($"    Jump if {(block.Conditional.JumpIfTrue ? "True" : "False")} ({conditionalBranch.Flags}) to Block[{index}]");
 
                     IOperation value = block.Conditional.Condition;
                     validateRoot(value);
-                    stringBuilder.Builder.AppendLine(OperationTreeVerifier.GetOperationTree(compilation, value, initialIndent: 8));
+                    stringBuilder.Append(OperationTreeVerifier.GetOperationTree(compilation, value, initialIndent: 8 + indent));
+                    validateBranch(block, conditionalBranch);
+                    stringBuilder.AppendLine();
+                }
+                else
+                {
+                    Assert.Null(conditionalBranch.Destination);
+                    validateBranch(block, conditionalBranch);
+                }
+
+                BasicBlock.Branch nextBranch = block.Next;
+
+                if (nextBranch.Destination != null || block.Kind != BasicBlockKind.Exit)
+                {
+                    if (nextBranch.Destination != null)
+                    {
+                        Assert.Same(blocks[nextBranch.Destination.Ordinal], nextBranch.Destination);
+                    }
+
+                    appendLine($"    Next ({nextBranch.Flags}) Block[{nextBranch.Destination?.Ordinal ?? (object)"null"}]");
+                }
+                else
+                {
+                    Assert.Equal(0, (int)nextBranch.Flags);
+                    Assert.Null(nextBranch.Destination);
                 }
 
-                void validateRoot(IOperation root)
+                validateBranch(block, nextBranch);
+
+                if (currentRegion.LastBlockOrdinal == block.Ordinal && i != blocks.Length - 1)
+                {
+                    leaveRegions(block.Region, block.Ordinal);
+                }
+                else
                 {
-                    visitor.Visit(root);
-                    Assert.Null(root.Parent);
-                    Assert.Null(((Operation)root).SemanticModel);
-                    Assert.True(CanBeInControlFlowGraph(root), $"Unexpected node kind OperationKind.{root.Kind}");
+                    lastPrintedBlockIsInCurrentReagion = true;
+                }
+
+            }
 
-                    foreach (var operation in root.Descendants())
+            regionMap.Free();
+            return pooledBuilder.ToStringAndFree();
+
+            PooledObjects.PooledDictionary<ControlFlowGraph.Region, int> buildRegionMap()
+            {
+                var result = PooledObjects.PooledDictionary<ControlFlowGraph.Region, int>.GetInstance();
+                int ordinal = 0;
```

Comment:
One more suggestion: would it make sense to use different identifiers for regions as for blocks? That could be done by having regions use a different range (100, 101, etc), or maybe converting region ordinals to letters (a, b, ...A, B, etc). 
The main benefit is to assist code reviewers using diff tools with identifier highlighting. When you select "2" it will show the other matching "2"'s. But there are two kinds of "2"s (blocks and regions)...